### PR TITLE
Added Xfs test to avocado[V4]

### PR DESCRIPTION
**Pull request by PraveenPenguin**
xfs test : use xfs test from git.kernel.org
	 : enable user to run single test
	 : enable user can run test in range
README file having how to use and what not to use

Signed-off-by: Praveen K Pandey <praveen@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
find job log in below link 

https://paste.fedoraproject.org/428761/74022036/

**Comment by PraveenPenguin** [comment]
v3#https://github.com/avocado-framework/avocado-misc-tests/pull/145

File: fs/xfstests.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        Source:
+        git://git.kernel.org/pub/scm/fs/xfs/xfstests-dev.git
+        '''
+
+        # Check for basic utilities
+        sm = SoftwareManager()
+        detected_distro = distro.detect()
+
+        packages = ['xfslibs-dev', 'uuid-dev', 'libtool-bin', 'e2fsprogs',
+                    'automake',  'gcc', 'libuuid1', 'quota', 'attr',
```
Comment (apahim) [inline fs/xfstests.py]: To much white-spaces between the automake and the gcc.

File: fs/xfstests.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        tests = [t.replace('.linux', '') for t in tests]
+        tests_list = [t[-7:-4] for t in tests if os.path.exists(t[:-4])]
+        tests_list.append('')
+        tests_list.sort()
+        return tests_list
+
+    def _run_sub_test(self, test):
+
+        os.chdir(self.srcdir)
+        self.log.info("Running test: %s" % test)
```
Comment (apahim) [inline fs/xfstests.py]: Please use this format instead: `self.log.info("Running test: %s", test)`

File: fs/xfstests.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+            raise self.error('Could not assert test success or failure, '
+                             'assuming failure. Please check debug logs')
+
+    def _get_groups(self):
+        '''
+        Returns the list of groups known to xfstests
+        By reading the group file and identifying unique mentions of groups
+        '''
+        groups = []
+        for l in open(os.path.join(self.srcdir, 'group')).readlines():
```
Comment (apahim) [inline fs/xfstests.py]: one-character variable names evil for the readability.

File: fs/xfstests.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+            m = self.GROUP_TEST_LINE_RE.match(l)
+            if m is not None:
+                groups = m.groups()[1].split()
+                for g in groups:
+                    if g not in groups:
```
Comment (apahim) [inline fs/xfstests.py]: l, m , g, ... please give them proper names :)

File: fs/xfstests.py [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+                        groups.add(g)
```
Comment (apahim) [inline fs/xfstests.py]: groups is a list... `add()` is not available for lists. You should use `append()` here.

File: fs/xfstests.py [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+        return groups
+
+    def _get_tests_for_group(self, group):
+        '''
+        Returns the list of tests that belong to a certain test group
+        '''
+        tests = []
+        for l in open(os.path.join(self.srcdir, 'group')).readlines():
+            m = self.GROUP_TEST_LINE_RE.match(l)
+            if m is not None:
```
Comment (apahim) [inline fs/xfstests.py]: again, l, m, ... proper names please.

File: fs/xfstests.py [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+    def setUp(self):
+        '''
+        Build xfstest
+        Source:
+        git://git.kernel.org/pub/scm/fs/xfs/xfstests-dev.git
+        '''
+
+        # Check for basic utilities
+        sm = SoftwareManager()
+        detected_distro = distro.detect()
```
Comment (apahim) [inline fs/xfstests.py]: This `detected_distro` is not being used anywhere.

File: fs/xfstests.py [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+
+        test_list = []
+
+        if self._test_range:
+            for item in self._test_range.split(','):
+                if '-' in item:
+                    start, end = item.split('-')
+                    test_list.extend(range(int(start), int(end) + 1))
+                else:
+                    test_list.append(int(item))
```
Comment (apahim) [inline fs/xfstests.py]: This block should be a helper function.

File: fs/xfstests.py (hunk 2 above)
Comment (PraveenPenguin) [inline fs/xfstests.py]: @apahim can you please help me what i need to do ?

Regards
Praveen

**Comment by PraveenPenguin** [comment]
@apahim thanks , included  suggestion in  PR #44 Please have a look

Regards
Praveen

**Pull request closed without merging**